clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -940,6 +943,99 @@ public void testUnsupportedForMessageFormatInTxnOffsetCommit() {
         assertFatalError(UnsupportedForMessageFormatException.class);
     }
 
+    @Test
+    public void testFencedInstanceIdInTxnOffsetCommit() {
```

Comment:
These test cases are a bit odd. Why would we get a `FENCED_INSTANCE_ID` if we have not provided an instance id? I think it would make sense to do all of the client changes in a single PR and all the server changes in a separate PR. I'm ok doing them all at once also since this PR is not that big.

Reply:
Cool, I will separate the PR to leave only server side changes